metadata_backend/helpers/workflow.py

```diff
@@ -0,0 +1,103 @@
+"""Utilities for processing a workflow."""
```

Comment:
i am starting to think based on what we do here, we might as well put the workflows in a collection, if need to check these many things on them.

For the schemas we did not need that much processing, that is why we don't have them in a mongodb collection, however i think for the workflows it might make sense. and yes we can use json schemas to validate them in mongodb https://www.mongodb.com/docs/manual/core/schema-validation/specify-json-schema/#std-label-schema-validation-json

This will eliminate the need of this custom class and processing we do on workflow JSONs in python.

For now just a thought.

Reply:
I don't see how it would reduce the need for processing.

One idea I had later was to create a single instance of each workflow.

The processing should happen only once, as the methods are marked as property, they should be cached by Python.

Reply:
to make it makes more sense to ask using a db query if objects of a certain schema are allowed multiple objects or not e.g.
we get a schema -> query.allowed_objects_workflow(schema)-> true/false

Reply:
but this is not the subject of this PR, was just an idea.

Reply:
Pushed some changes. What do you think?